src/uct/ib/mlx5/rc/rc_mlx5_iface.c

```diff
@@ -417,10 +416,10 @@ static ucs_status_t uct_rc_mlx5_iface_preinit(uct_rc_mlx5_iface_common_t *iface,
 #endif
     ucs_status_t status;
 
-    status = uct_rc_mlx5_iface_parse_srq_topo(md, mlx5_config,
-                                              &iface->config.srq_topo,
-                                              !!(init_attr->flags &
-                                                 UCT_IB_DDP_SUPPORTED));
+    status = uct_rc_mlx5_iface_parse_srq_topo(
+            md, mlx5_config, &iface->config.srq_topo,
+            (init_attr->flags & UCT_IB_DDP_SUPPORTED) &&
+                    (mlx5_config->ddp_enable != UCS_NO));
```

Comment:
1. maybe just pass mlx5_config to uct_rc_mlx5_iface_parse_srq_topo
2. i think it would be better to set srq topology according to the same logic we set iface->config.dp_ordering